include/swift/AST/GenericSignature.h

```diff
@@ -218,10 +218,13 @@ class alignas(1 << TypeAlignInBits) GenericSignature final
   /// Canonicalize the components of a generic signature.
   CanGenericSignature getCanonicalSignature() const;
 
-  /// Create a new generic environment that provides fresh contextual types
+  /// Retrieve the generic signature builder for the given generic signature.
+  GenericSignatureBuilder *getGenericSignatureBuilder();
+
+  /// Returns the generic environment that provides fresh contextual types
   /// (archetypes) that correspond to the interface types in this generic
   /// signature.
-  GenericEnvironment *createGenericEnvironment();
+  GenericEnvironment *getGenericEnvironment();
```

Comment:
Should this be `const` (and the field `mutable`)? Although I guess all of GenericSignature is consty, really.

Reply:
I sort of prefer to use const_cast instead, because then the point of mutation is explicit...